src/classes/MergeService.cls

```diff
@@ -16,6 +17,7 @@ global with sharing class MergeService {
      * As of API 40.0 the only supported objects are Lead, Account and Contact.
      */
     private String sobjectName { get; set; }
+    private String sobjectNameOriginal { get; set; } // added by danhowellnz
```

Comment:
Totally agree. This was more of a proof of concept then a polished product. I didn't end up using merging in my user case.